src/screens/Search/Lists/AlgoResultsList.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { View, FlatList, Text, TouchableOpacity } from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import ProgressiveImage from 'kitsu/components/ProgressiveImage';
+
+const ResultsList = ({
+  hits,
+  hasMore,
+  refine,
+  onPress,
+  scrollEnabled = true,
+  numColumns = 4,
+  imageSize = { h: 125, w: 91 },
+}) => {
+  const onEndReached = () => {
+    if (hasMore) {
+      refine();
+    }
+  };
+  return (
+    <View style={{ backgroundColor: '#FAFAFA' }}>
+      <FlatList
+        removeClippedSubviews={false}
+        data={hits}
+        onEndReached={onEndReached}
+        onEndReachedThreshold={0.5}
+        getItemLayout={(data, index) => ({
+          length: imageSize.h,
+          offset: imageSize.h * index,
+          index,
+        })}
+        initialNumToRender={10}
+        numColumns={numColumns}
+        scrollEnabled={scrollEnabled}
+        contentContainerStyle={styles.list}
+        renderItem={e => renderItem(e, imageSize, onPress)}
+      />
+    </View>
+  );
+};
+
+const renderItem = ({ item }, imageSize, onPress) => {
+  let title = null;
+  if (item.titles) {
+    title = item.titles.en || item.titles.en_jp;
+  }
+  const { h, w } = imageSize;
+  const m = imageSize.m || 1;
+  return (
+    <TouchableOpacity onPress={() => onPress(item)}>
+      <View
+        style={{
+          height: h - m * 2,
+          width: w - m * 2,
+          margin: m,
+        }}
+        onPress={() => onPress(item)}
+      >
+        <ProgressiveImage
+          onPress={() => onPress(item)}
+          source={{ uri: (item.posterImage || {}).small }}
+          style={{
+            height: h - m * 2,
+            width: w - m * 2,
+          }}
+        />
+        {title &&
+          <LinearGradient colors={['transparent', 'black']} style={styles.linearGradient}>
+            <Text
+              style={{
+                color: 'white',
```

Comment:
move this object to styles? you could also import `commonStyles` and do:

`style={[commonStyles.text, commonStyles.colorWhite, styles.titleText]}`